### PR TITLE
Upgrade Bourbon to 5.1.0 and fix deprecation warnings

### LINKED ISSUE #1189: Hundreds of Bourbon deprecation warnings
Repro:
`rm -rf node_modules && make deps && make generate`

There are hundreds of warnings looking like this:
```
WARNING: [Bourbon] [Deprecation] `placeholder` is deprecated and will be removed in 5.0.0. We suggest using an automated prefixing tool, like Autoprefixer.
Backtrace:
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:10, in mixin `-bourbon-deprecate`
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:17, in mixin `-bourbon-deprecate-for-prefixing`
	node_modules/bourbon/app/assets/stylesheets/css3/_placeholder.scss:2, in mixin `placeholder`
	frontend/styles/global/_forms.scss:2

WARNING: [Bourbon] [Deprecation] `display` is deprecated and will be removed in 5.0.0. We suggest using an automated prefixing tool, like Autoprefixer.
Backtrace:
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:10, in mixin `-bourbon-deprecate`
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:17, in mixin `-bourbon-deprecate-for-prefixing`
	node_modules/bourbon/app/assets/stylesheets/css3/_flex-box.scss:87, in mixin `display`
	frontend/styles/global/_global.scss:86

WARNING: [Bourbon] [Deprecation] `flex-grow` is deprecated and will be removed in 5.0.0. We suggest using an automated prefixing tool, like Autoprefixer.
Backtrace:
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:10, in mixin `-bourbon-deprecate`
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:17, in mixin `-bourbon-deprecate-for-prefixing`
	node_modules/bourbon/app/assets/stylesheets/css3/_flex-box.scss:213, in mixin `flex-grow`
	frontend/styles/global/_global.scss:91

WARNING: [Bourbon] [Deprecation] `align-self` is deprecated and will be removed in 5.0.0. We suggest using an automated prefixing tool, like Autoprefixer.
Backtrace:
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:10, in mixin `-bourbon-deprecate`
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:17, in mixin `-bourbon-deprecate-for-prefixing`
	node_modules/bourbon/app/assets/stylesheets/css3/_flex-box.scss:290, in mixin `align-self`
	frontend/styles/global/_global.scss:92

WARNING: [Bourbon] [Deprecation] `display` is deprecated and will be removed in 5.0.0. We suggest using an automated prefixing tool, like Autoprefixer.
Backtrace:
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:10, in mixin `-bourbon-deprecate`
	node_modules/bourbon/app/assets/stylesheets/_bourbon-deprecate.scss:17, in mixin `-bourbon-deprecate-for-prefixing`
	node_modules/bourbon/app/assets/stylesheets/css3/_flex-box.scss:87, in mixin `display`
	frontend/components/AuthenticationFormWrapper/_styles.scss:2
```

### PR DESCRIPTION
**Pull request by zwass**
- Replace uses of deprecated Bourbon helpers with raw CSS
- Add autoprefixer into the build chain to prefix the now removed helpers

This process was achieved by running through each of the deprecation warnings and using the following bash function to replace it in all files:

```
function bourbon-deprecate() {
    grep -rl "@include $1" ./frontend --exclude-dir=.git | xargs sed -i '' -E "s/@include $1[(](.*)[)]/$1: \1/g"
}
```

For some helpers, this did not result in valid CSS, so manual modifications were made.

Closes #1189 #1274

### CODE REVIEW

**Comment by zwass** [comment]
This took some fiddling, but I believe the upgrade is complete with full functionality.

File: frontend/components/StackedWhiteBoxes/_styles.scss [hunk 1]
```diff
@@ -1,21 +1,21 @@
 .stacked-white-boxes {
   @include position(absolute, null null null 50%);
-  @include transition(all 300ms ease-in);
+  transition: all 300ms ease-in;
```
Comment (kyleknighted) [inline frontend/components/StackedWhiteBoxes/_styles.scss]: i think we should do 
```suggestion
  transition: transform 300ms ease-in;
```

File: frontend/components/forms/LoginForm/_styles.scss [hunk 2]
```diff
@@ -1,24 +1,23 @@
 .login-form {
-  @include transition(all 300ms ease-in);
-  @include transform(scale(1));
+  transition: all 300ms ease-in;
```
Comment (kyleknighted) [inline frontend/components/forms/LoginForm/_styles.scss]:
```suggestion
  transition: transform 300ms ease-in;
```

File: frontend/components/UserBlock/_styles.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
 .user-block {
-  @include transition(all 300ms ease-in-out);
+  transition: all 300ms ease-in-out;
```
Comment (kyleknighted) [inline frontend/components/UserBlock/_styles.scss]:
```suggestion
  transition: border 300ms ease-in-out, background-color 300ms ease-in-out;
```

File: frontend/components/UserBlock/_styles.scss [hunk 4]
```diff
@@ -20,7 +20,7 @@
   }
 
   &__header {
-    @include transition(all 300ms ease-in-out);
+    transition: all 300ms ease-in-out;
```
Comment (kyleknighted) [inline frontend/components/UserBlock/_styles.scss]:
```suggestion
  transition: color 300ms ease-in-out, background-color 300ms ease-in-out;
```

File: frontend/components/UserBlock/_styles.scss [hunk 5]
```diff
@@ -73,7 +73,7 @@
 
   &__avatar {
     @include size(100px);
-    @include transition(all 300ms ease-in-out);
+    transition: all 300ms ease-in-out;
```
Comment (kyleknighted) [inline frontend/components/UserBlock/_styles.scss]:
```suggestion
    transition: border 300ms ease-in-out;
```

File: frontend/components/UserBlock/_styles.scss [hunk 6]
```diff
@@ -154,7 +154,7 @@
   }
 
   &__email {
-    @include transition(all 300ms ease-in-out);
+    transition: all 300ms ease-in-out;
```
Comment (kyleknighted) [inline frontend/components/UserBlock/_styles.scss]:
```suggestion
    transition: color 300ms ease-in-out;
```

File: frontend/pages/LoginPage/_styles.scss [hunk 7]
```diff
@@ -1,10 +1,10 @@
 .login-form-animation-leave {
-  @include transform(scale(1));
+  transform: scale(1);
   opacity: 1;
 
   &.login-form-animation-leave-active {
-    @include transition(all 300ms ease-in);
-    @include transform(scale(1.3));
+    transition: all 300ms ease-in;
```
Comment (kyleknighted) [inline frontend/pages/LoginPage/_styles.scss]:
```suggestion
    transition: opacity 300ms ease-in, transform 300ms ease-in;
```

File: .sass-lint.yml [hunk 8]
```diff
@@ -83,7 +83,7 @@ rules:
       allow-element-with-class: false
       allow-element-with-id: false
   no-trailing-zero: 1
-  no-transition-all: 1
+  no-transition-all: 0
```
Comment (kyleknighted) [inline .sass-lint.yml]:
```suggestion
  no-transition-all: 1
```

**Comment by kyleknighted** [comment]
We should follow this up with removal of Bourbon all-together

**Comment by zwass** [comment]
I'd be thrilled to remove Bourbon altogether if you think that's something we can do.

**Pull request merged**